DESCRIPTION.md

```diff
@@ -8,7 +8,7 @@ Source code is also available at: https://github.com/snowflakedb/snowflake-conne
 
 # Release Notes
 
-- v3.12.0(TBD)
+- v3.12.0(July 26,2024)
```

Comment:
Maybe we could group these into updates and bug fixes. Not necessarily create headings, but at least they would show together in the list.